tsdb/engine/tsm1/cache.go

```diff
@@ -219,7 +240,10 @@ func (c *Cache) Write(key string, values []Value) error {
 		return ErrCacheMemoryExceeded
 	}
 
-	c.write(key, values)
+	if err := c.write(key, values); err != nil {
+		c.mu.Unlock()
```

Comment:
Should we update `c.stats.WriteErr` or maybe add a new stat for this unusual condition?